lib/src/main/kotlin/ca/aqtech/mctreesearch4j/AdvancedStatelessSolver.kt

```diff
@@ -3,7 +3,7 @@ package ca.aqtech.mctreesearch4j
 import kotlin.math.ln
 import kotlin.math.sqrt
 
-open class AdvancedStatelessSolver<TState, TAction>(
+open class AdvancedStatelessSolver<TState, TAction> (
```

Comment:
wouldn't this be `<StateType, ActionType>`

Reply:
Actually I want to get your take. I don't see this used anywhere so can we delete this? 

Personally I feel the functionality here seems too specialized. It was for collecting some trends but I think theoretically users may want to collect all kinds of information so unless we can say this is a very common diagnostic to collect, I'd rather this get removed from the lib package.

Reply:
Key reason 1 i wanted to keep `AdvancedStatelessSolver` was because I had difficulty writing the extension in Scala, though it could be done in principle. Key reason 2, is this could be a way to demonstrate inheritance from the abstract solver classes to make new solvers.

If you could make this Scala extension work as intended, then I'd remove the `AdvancedStatelessSolver`
https://github.com/larkz/connect4-scala/blob/master/src/main/scala/connect4/Connect4Solver.scala

Reply:
> because I had difficulty writing the extension in Scala

Sounds like a reason to make changes to the library itself to make it easy to write subclasses. If our library can't be extended, then we messed up. We shouldn't make do by shipping something like `AdvancedStatelessSolver` to address one particular use case.

> this could be a way to demonstrate inheritance from the abstract solver classes to make new solvers.

For demonstration purposes, I think `ReversiSolverHeuristicSim` is a better example: https://github.com/JunTaoLuo/mctreesearch4j/blob/main/app/src/main/kotlin/Reversi/ReversiSolverHeuristicSim.kt. It shows a working sample of improving the base solvers and we actually have data to back it up.

Reply:
As for scala, I'm not familiar with the language or how it's hooked up to use our library. Can you help walk me through it in our next sync?